From 58cfdc8020d924a908c4eebaf0a900745579ce3f Mon Sep 17 00:00:00 2001
From: Drew Fustini <drew@beagleboard.org>
Date: Sun, 30 May 2021 12:50:09 -0700
Subject: [PATCH 46/56] gpio: starfive-jh7100: rename from VIC to JH7100

VIC was a StarFive internal project name, thus rename to JH7100 which is
the public SoC name [1]. This is SoC is used on the BeagleV Starlight
JH7100 board [2].

[1] https://github.com/starfive-tech/beaglev_doc
[2] https://github.com/beagleboard/beaglev-starlight

Signed-off-by: Drew Fustini <drew@beagleboard.org>
---
 drivers/gpio/Kconfig                                      | 6 +++---
 drivers/gpio/Makefile                                     | 2 +-
 .../gpio/{gpio-starfive-vic.c => gpio-starfive-jh7100.c}  | 8 +++-----
 3 files changed, 7 insertions(+), 9 deletions(-)
 rename drivers/gpio/{gpio-starfive-vic.c => gpio-starfive-jh7100.c} (99%)

diff --git a/drivers/gpio/Kconfig b/drivers/gpio/Kconfig
index 32f3a4a2eeef..26630e4852c0 100644
--- a/drivers/gpio/Kconfig
+++ b/drivers/gpio/Kconfig
@@ -542,13 +542,13 @@ config GPIO_SIFIVE
 	help
 	  Say yes here to support the GPIO device on SiFive SoCs.
 
-config GPIO_STARFIVE_VIC
-	bool "Starfive VIC GPIO support"
+config GPIO_STARFIVE_JH7100
+	bool "StarFive JH7100 GPIO support"
 	depends on OF_GPIO
 	select GPIOLIB_IRQCHIP
 	default y if SOC_STARFIVE_VIC7100
 	help
-	  Say yes here to support the GPIO device on Starfive VIC SoCs.
+	  Say yes here to support the GPIO device on StarFive JH7100 SoC.
 
 config GPIO_SIOX
 	tristate "SIOX GPIO support"
diff --git a/drivers/gpio/Makefile b/drivers/gpio/Makefile
index cb5dd75e61f5..939922eaf5f3 100644
--- a/drivers/gpio/Makefile
+++ b/drivers/gpio/Makefile
@@ -132,7 +132,7 @@ obj-$(CONFIG_GPIO_SAMA5D2_PIOBU)	+= gpio-sama5d2-piobu.o
 obj-$(CONFIG_GPIO_SCH311X)		+= gpio-sch311x.o
 obj-$(CONFIG_GPIO_SCH)			+= gpio-sch.o
 obj-$(CONFIG_GPIO_SIFIVE)		+= gpio-sifive.o
-obj-$(CONFIG_GPIO_STARFIVE_VIC)		+= gpio-starfive-vic.o
+obj-$(CONFIG_GPIO_STARFIVE_JH7100)	+= gpio-starfive-jh7100.o
 obj-$(CONFIG_GPIO_SIOX)			+= gpio-siox.o
 obj-$(CONFIG_GPIO_SL28CPLD)		+= gpio-sl28cpld.o
 obj-$(CONFIG_GPIO_SODAVILLE)		+= gpio-sodaville.o
diff --git a/drivers/gpio/gpio-starfive-vic.c b/drivers/gpio/gpio-starfive-jh7100.c
similarity index 99%
rename from drivers/gpio/gpio-starfive-vic.c
rename to drivers/gpio/gpio-starfive-jh7100.c
index 484677c1eb43..f9d255c95d6d 100755
--- a/drivers/gpio/gpio-starfive-vic.c
+++ b/drivers/gpio/gpio-starfive-jh7100.c
@@ -47,8 +47,6 @@
 
 #define MAX_GPIO	 64
 
-#define PROC_VIC "vic_gpio"
-
 struct starfive_gpio {
 	raw_spinlock_t		lock;
 	void __iomem		*base;
@@ -304,7 +302,7 @@ static void starfive_irq_disable(struct irq_data *d)
 }
 
 static struct irq_chip starfive_irqchip = {
-	.name		= "starfive-gpio",
+	.name		= "starfive-jh7100-gpio",
 	.irq_set_type	= starfive_irq_set_type,
 	.irq_mask	= starfive_irq_mask,
 	.irq_unmask	= starfive_irq_unmask,
@@ -541,7 +539,7 @@ static const struct of_device_id starfive_gpio_match[] = {
 static struct platform_driver starfive_gpio_driver = {
 	.probe	= starfive_gpio_probe,
 	.driver	= {
-		.name = "starfive_gpio",
+		.name = "gpio_starfive_jh7100",
 		.of_match_table = of_match_ptr(starfive_gpio_match),
 	},
 };
@@ -560,4 +558,4 @@ module_exit(starfive_gpio_exit);
 
 MODULE_LICENSE("GPL");
 MODULE_AUTHOR("Huan Feng <huan.feng@starfivetech.com>");
-MODULE_DESCRIPTION("Starfive VIC GPIO generator driver");
+MODULE_DESCRIPTION("StarFive JH7100 GPIO driver");
-- 
2.30.2

